README.md

```diff
@@ -1,3 +1,125 @@
-# Safe wrappers playground
+# **AA Wrap Demo** &middot; [![license](https://img.shields.io/badge/license-MIT-blue)](https://shields.io)
 
-Playground to demostrate the usage of safe wrappers.
\ No newline at end of file
+<!-- This demo showcases how the AA Wrap can be used to integrate account abstraction functionality into different deployment platforms – web browsers, mobile, gaming and beyond. -->
+
+<img src="images/multiplatform.png" width="500x">
+
+## What's the AA Wrap?
+
+The AA Wrap is a Polywrap-powered version of the [Safe Account Abstraction (AA) SDK](https://docs.safe.global/learn/safe-core/safe-core-account-abstraction-sdk).
+
+The Safe AA SDK lets developers build ["smart accounts"](https://docs.safe.global/learn/what-is-a-smart-contract-account), accounts with fully customizable functionality. Smart accounts aim to bring much-needed UX improvements to web3.
+
+Today, the AA Wrap includes functionality to:
+
+- create customizable, secure, and modular smart accounts.
+- enable a gas-less user experience via Gelato, offer sponsored transactions, and allow for fee payments in ERC-20 tokens.
+
+## Demo Overview
+
+In this demo, you'll be executing scripts that showcase the AA Wrap's main value props: **multi-platform** and **composability**.
+
+<img src="images/fetch.png" width="300x">
+
+1. **Multi–platform** – The same AA Wrap is fetched from IPFS and then integrated into two different programming environments: JavaScript and Rust.
+
+2. **Composability** – The AA Wrap's functionality can be easily extended by composing it with other Wraps. The current AA Wrap is composed of 3 separate Wraps: Ethers, Gelato Relay and Safe{Core}.
+
+## Demo Setup
+
+First, you'll set up the demo.
+
+1. Change directory to `js`:
+
+```
+cd js
+```
+
+2. Update the salt nonce in `deploy-safe.ts` to `0x` followed by a few randomly chosen digits. For example,
+
+This salt nonce is used later to create the new Safe address.
+
+```ts
+const SALT_NONCE = "0x185593";
+```
+
+3. Run the `deploy-safe.ts` script to deploy the Safe smart account.
+
+```
+yarn deploy
+```
+
+> For the purposes of this demo, we're deploying to the Goerli test network. You can view your Safe on [Goerli Etherscan block explorer](https://goerli.etherscan.io/).
+
+4. Create a `.env` file at the root of the repo and add in the Safe address:
```

Comment:
we need to put the creation of `.env` as first step, being the main reason that the user needs to define a private key, which will take care of paying for the deployment of the safe